File: recTecm/whichOut.m
function [idx, threArry2] = whichOut (timeArry, timestamp)
    threArry1 = exp(1./(timestamp - timeArry));
    buff      = - exp (- threArry1);
    threArry2 = 1 - exp(buff); % the out threshold. 
    probArry  = rand(size(timeArry, 1), 1);
    
    idx = find((threArry2-probArry) == max(threArry2 - probArry));
    
    
    
end

File: recTecp/whichOut.py
import numpy as np
import math

def whichOut(timeArray, timestamp):
    threArray1 = np.exp(1/(timestamp - timeArray))
    buff = - np.exp(-threArray1)
    threArray2 = 1 - np.exp(buff)
    probArry = np.random.uniform(0, 1, len(timeArray))
    right = threArray2 - probArry
    index = np.where(right == right.max())
    return index[0][0]
